File: 15주차_프로그래머스_lv1/159994.py
def solution(cards1, cards2, goal):
    answer = ''
    idx1 = 0
    idx2 = 0
    for i in range(len(goal)):
        if goal[i] == cards1[idx1]:
            if idx1 < len(cards1) - 1:
                idx1 += 1
        elif goal[i] == cards2[idx2]:
            if idx2 < len(cards2) - 1:
                idx2 += 1
        else:
            return 'No'
    return 'Yes'

File: 2025/03/159994.js
function solution(cards1, cards2, goal) {
  var answer = "Yes";
  let idx = 0;
  while (idx < goal.length) {
    let flag = false;
    if (cards1.length > 0 && cards1[0] === goal[idx]) {
      cards1.shift();
      flag = true;
      idx += 1; // 다음 goal 문자열을 비교한다.
    } else if (cards2.length > 0 && cards2[0] === goal[idx]) {
      cards2.shift();
      flag = true;
      idx += 1;
    }

    if (idx < goal.length && flag === false) {
      answer = "No";
      break;
    }
  }
  return answer;
}
//프로그래머스 연습문제
